### PR TITLE
fix(toast-notification): remove default value from optional caption prop

### PR DESCRIPTION
**Pull request by janhassel**
According to the toast notification's [component guidelines](https://www.carbondesignsystem.com/components/notification/usage/#toast-notifications) and the propTypes, the `caption` prop (intended for timestamp) is optional. However, currently it defaults to "provide a caption" which
a) makes it seem to be required (as other component usually do this for missing, required props such as titles), and
b) require the developer to pass an empty caption to the component in order to remove it, which is unusual for an optional prop

#### Changelog

**Removed**

- removed default value ("provide a caption") for `props.caption` in `ToastNotification`

#### Testing / Reviewing

- If no caption is passed, there should be no rendered caption
- If a caption is passed, it should be rendered

### CODE REVIEW

File: packages/react/src/components/Notification/Notification.js [hunk 1]
```diff
@@ -399,7 +399,7 @@ ToastNotification.propTypes = {
 ToastNotification.defaultProps = {
   kind: 'error',
   title: 'provide a title',
-  caption: 'provide a caption',
+  caption: '',
```
Comment (joshblack) [inline packages/react/src/components/Notification/Notification.js]: Would it make sense to remove the `caption` as a default prop?

Comment (janhassel) [reply packages/react/src/components/Notification/Notification.js]: Yeah, good point!

**Comment by tw15egan** [comment]
Everything looks good code-wise, just want to get some design input if the padding is correct when there is no caption 

Screen Shot 2021-02-25 at 9 56 13 AM

**Review by jnm2377 — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Review by kingtraceyj — CHANGES_REQUESTED** [review]
the padding from the bottom is looking a little large. it should be 16px.
current:
image

updated:
image

**Comment by janhassel** [comment]
@kingtraceyj updated 👍

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Review by kingtraceyj — APPROVED** [review]
looks good to me. thanks for the update!

**Pull request merged**